ui/src/data/DataStore.ts

```diff
@@ -0,0 +1,165 @@
+  const zomeApi = new ZomeApi(appWs);
+  setAgentPubKey(agentPubKey);
+  setCellId(app_info.cell_data[0].cell_id);
+  setZomeApi(zomeApi);
+  dataStore = new DataStore();
+  await dataStore.fetchOrCreateRoot();
+}
+
+/**
+ * Fetches the DataStore or initialtiazes a new one
+ * @returns DataStore
+ */
+export default function getDataStore(): DataStore {
+  if (dataStore == undefined) {
+    dataStore = new DataStore();
+  }
+  return dataStore;
+}
+
+export class DataStore extends DataStoreBase {
+
+  constructor() {
+    super();
+  }
+
+  // Root helpers
+
+  /**
+   * Checks to see if we have anything in our DHT and chain, if not sets it up.
+   */
+  public override async fetchOrCreateRoot() {
+    // check if root object exists
+    const res = await this.zomeApi.get_thing('root');
+    if (res[0].val.data === '') {
+      // if it doesn't, create it and a placeholder plan
+      console.log('root does not exist. creating...');
+      const plan = new Plan({
+        name: 'Default Plan Name'
+      });
+      this.root = new Root({planId: plan.id})
+      await this.put(this.root);
+      await this.set(plan);
+    } else  {
+      // We have the data, lets hydrate it
+      this.hydrateFromZome(res);
+    }
+  }
+
+  // ProcessSprecification helpers
+
+  // Get from store
+  // Equivalent to getCursor(ProcessSpecification.getPath(id))
+  public getProcessSpecification(id: string): ProcessSpecification {
+    return this.root.processSpecification[id];
+  }
+
+  // For fetching updates when we get signals
+  public async fetchProcessSpecification(id: string) {
+    return await this.fetchSingle(ProcessSpecification.getPath(id));
+  }
+
+  public getProcessSpecifications(): Array<ProcessSpecification> {
+    return Object.values(this.root.processSpecification);
+  }
+
+  public async fetchProcessSpecifications() {
+    return this.fetchAll(ProcessSpecification.getPrefix());
+  }
+
+
+  // ResourceSpecification helpers
+
+  public getResourceSpecification(id: string): ResourceSpecification {
+    return this.root.resourceSpecification[id];
+  }
+
+  public async fetchResourceSpecification(id: string) {
+    return await this.fetchSingle(ResourceSpecification.getPath(id));
+  }
+
+  public getResourceSpecifications(): Array<ResourceSpecification> {
+    return Object.values(this.root.resourceSpecification);
+  }
+
+  public async fetchResourceSpecifications() {
+    return await this.fetchAll(ResourceSpecification.getPrefix());
+  }
+
+
+  // Agent helpers
+
+  public getAgent(id: string): Agent {
+    return this.root.agent[id];
+  }
+
+  public async fetchAgent(id: string): Promise<Agent> {
+    return await this.fetchSingle(Agent.getPath(id)) as Agent;
+  }
+
+  public getAgents(): Array<Agent> {
+    return Object.values(this.root.agent);
+  }
+
+  public async fetchAgents() {
+    return await this.fetchAll(Agent.getPrefix());
+  }
+
+
+  // Plan helpers
+  public getPlan(id: string): Plan {
```

Comment:
Musing for the day: Make id optional. If it is not provided, default to the current planId stored in root.

Reply:
I actually think we shouldn't. Mostly because at some point we will need to have application state for which plan is being looked at. At a higher level, we could have views of multiple plans and how they connect to each other, and then we could zoom in on one specific plan.